internal/api/registry.go

```diff
@@ -17,12 +17,49 @@ package api
 import (
 	"fmt"
 	"sync"
+	"time"
 
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/utils/set"
 
 	azcorearm "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
 )
 
+var (
+	// we have some private fields that cmp.Diff chokes on. This skips them.
+	// Use only for error messages unless you know what you're doing.
+	CmpDiffOptions = []cmp.Option{
+		cmpopts.IgnoreFields(azcorearm.ResourceID{}, "ResourceType"),
+		cmpopts.IgnoreFields(azcorearm.ResourceID{}, "isChild"),
+		cmpopts.IgnoreFields(azcorearm.ResourceID{}, "stringValue"),
+		cmpopts.IgnoreFields(InternalID{}, "path"),
+		cmpopts.IgnoreFields(InternalID{}, "kind"),
```

Comment:
Would [cmpopts.IgnoreUnexported](https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#IgnoreUnexported) work here instead of enumerating them all?

Reply:
It doesn't work.  I think it's because the type is aliased to an internal type perhaps?